pcap-int.h

```diff
@@ -107,6 +107,11 @@ extern int pcap_new_api;
  */
 extern int pcap_utf_8_mode;
 
+/*
+ * Map packet buffers below 2GB on Linux.
```

Comment:
Wine isn't Linux-only, so this could conceivably be an issue on other platforms.

I might change the comment to "Map packet buffers with 32-bit addresses".

Reply:
Furthermore, if memory-mapped capture *isn't* being done, would it be necessary to allocate the buffer into which packets are read (e.g., with BPF devices) at a sufficiently-low address (as per [Microsoft's documentation of the address space layout in 32-bit Windows](https://learn.microsoft.com/en-us/windows/win32/memory/virtual-address-space))?

Reply:
> Wine isn't Linux-only, so this could conceivably be an issue on other platforms. 
> I might change the comment to "Map packet buffers with 32-bit addresses".

Thanks for the review. True, it would be nice to support this on macOS too, for example, but it's 
not straightforward there.

In the default 32-bit address layout on Windows the low 2GB is reserved for user space and the system uses the high 2GB. This maps nicely to the MAP_32BIT flag on Linux which enforces the same boundary. macOS does have a MAP_32BIT flag but it enforces a 4GB boundary, which may upset some Windows applications.

There's a non-default 3-1 split on Windows where apps can use 3GB, provided they set the Large Address Aware flag. They could still get confused by addresses reserved for the system but in practice this may work well enough, so it may still be worth supporting this scenario.

However, it's not as simple as passing the MAP_32BIT flag because 64-bit macOS maps all of the first 4GB to the zero page, making it inaccessible by default. We'd need to pass a
linker flag (-pagezero_size) to work around this limitation.

Reply:
> Furthermore, if memory-mapped capture _isn't_ being done, would it be necessary to allocate the buffer into which packets are read (e.g., with BPF devices) at a sufficiently-low address (as per [Microsoft's documentation of the address space layout in 32-bit Windows](https://learn.microsoft.com/en-us/windows/win32/memory/virtual-address-space))?

I'm not sure how you would allocate the buffer below e.g. the 2GB limit without using mmap(). It probably makes sense to return an error when PCAP_MMAP_32BIT can't be supported, so that the caller can fall back to a different strategy.